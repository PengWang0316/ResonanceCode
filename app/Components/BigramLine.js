import React from 'react';
import PropTypes from 'prop-types';

const basicStyles = { height: '3px', background: '#000' };

const BigramLine = ({ isBroken }) => {
  if (isBroken) return (
    <div className="d-flex" style={{ width: '30px' }}><div style={{ ...basicStyles, width: '45%' }} /><div style={{ ...basicStyles, width: '10%', background: '#FFF' }} /><div style={{ ...basicStyles, width: '45%' }} /></div>
  );
  return <div style={{ ...basicStyles, width: '30px' }} />;
};
BigramLine.propTypes = { isBroken: PropTypes.bool };
BigramLine.defaultProps = { isBroken: false };
export default BigramLine;
