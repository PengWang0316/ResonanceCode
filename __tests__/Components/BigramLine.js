import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import BigramLine from '../../app/Components/BigramLine';

describe('BigramLine test', () => {
  // const defaultProps = {};
  // const getShallowComponent = (props = defaultProps) => shallow(<BigramLine {...props} />);

  test('BigramLine is not broken snapshot', () => expect(renderer.create(<BigramLine />).toJSON()).toMatchSnapshot());

  test('BigramLine is broken snapshot', () => expect(renderer.create(<BigramLine isBroken />).toJSON()).toMatchSnapshot());
});
