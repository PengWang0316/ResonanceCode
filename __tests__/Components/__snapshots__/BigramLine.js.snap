// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigramLine test BigramLine is broken snapshot 1`] = `
<div
  className="d-flex"
  style={
    Object {
      "width": "30px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#000",
        "height": "3px",
        "width": "45%",
      }
    }
  />
  <div
    style={
      Object {
        "background": "#FFF",
        "height": "3px",
        "width": "10%",
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "height": "3px",
        "width": "45%",
      }
    }
  />
</div>
`;

exports[`BigramLine test BigramLine is not broken snapshot 1`] = `
<div
  style={
    Object {
      "background": "#000",
      "height": "3px",
      "width": "30px",
    }
  }
/>
`;
